oe_authentication.module

```diff
@@ -10,10 +10,19 @@ declare(strict_types=1);
 /**
  * Implements hook_user_cancel_methods_alter().
  *
- * Remove the option to delete users while keeping the option to block them.
+ * Conditionally removes the option to delete users.
  */
 function oe_authentication_user_cancel_methods_alter(&$methods) {
-  if (\Drupal::currentUser()->id() == 1) {
+  $current_user = \Drupal::currentUser();
+  $restrict_user_delete = \Drupal::configFactory()
+    ->get('oe_authentication.settings')
+    ->get('restrict_user_delete_cancel_methods');
```

Comment:
We forgot to add a fallback to TRUE, just for the sake of it.

Reply:
I thought we don't, and rely on upgrade path?
I don't mind if we add it..

Reply:
I don't know we went back and forth a lot on this. I first said no, then I said ok to add it just to avoid any disruption.